packages/@aws-cdk/aws-batch/lib/job-definition.ts

```diff
@@ -426,7 +498,26 @@ export class JobDefinition extends Resource implements IJobDefinition {
     return vars;
   }
 
-  private buildJobContainer(container?: JobDefinitionContainer): CfnJobDefinition.ContainerPropertiesProperty | undefined {
+  /**
+   * Validates the properties provided for a new job definition.
+   */
+  private validateProps(props: JobDefinitionProps) {
+    if (props === undefined) {
+      return;
+    }
+
+    if (props.platformCapabilities !== undefined && props.platformCapabilities.includes(PlatformCapabilities.FARGATE)
+      && props.container.executionRole === undefined) {
+      throw new Error('Fargate job must have executionRole set');
+    }
+
+    if (props.platformCapabilities !== undefined && props.platformCapabilities.includes(PlatformCapabilities.EC2)
```

Comment:
If `props.platformCapabilities` is undefined, it defaults to `EC2` on the service side, so if `networkConfiguration` is specified, this would presumably result in a deployment failure right? 

So `undefined` in this case should be treated the same as `EC2`...

Reply:
Yes, updated